ui-v2/src/routes/concurrency-limits.tsx

```diff
@@ -1,9 +1,35 @@
+import {
+	buildListGlobalConcurrencyLimitsQuery,
+	useListGlobalConcurrencyLimitsSuspense,
+} from "@/hooks/global-concurrency-limits";
 import { createFileRoute } from "@tanstack/react-router";
+import { zodSearchValidator } from "@tanstack/router-zod-adapter";
+import { z } from "zod";
+
+/**
+ * Schema for validating URL search parameters for the variables page.
+ * @property {number} tab - The number of items to skip (for pagination). Must be non-negative. Defaults to 0.
+ */
+const searchParams = z
+	.object({
+		tab: z.enum(["Global", "Task Run"]).default("Global"),
+	})
+	.strict();
 
 export const Route = createFileRoute("/concurrency-limits")({
+	validateSearch: zodSearchValidator(searchParams),
 	component: RouteComponent,
+	wrapInSuspense: true,
+	loader: ({ context: { queryClient } }) =>
+		queryClient.ensureQueryData(buildListGlobalConcurrencyLimitsQuery()),
 });
 
 function RouteComponent() {
+	const { tab } = Route.useSearch();
+	const { data } = useListGlobalConcurrencyLimitsSuspense();
+
+	// TODO: pass values down to Page component
+	console.log({ tab, data });
+
```

Comment:
@desertaxle This is feels a bit weird from this discussion: https://github.com/PrefectHQ/prefect/pull/16220#discussion_r1870220994

If we're already passing `queryOptions` for the loader, it feels more natural to just call a `useSuspenseQuery` with the same `queryOption` down in the `RouteComponent` (L30). 

It feels a little bit more flexible, especially if we'll need to evolve from `useSuspenseQuery` to `useSuspenseQueries` for multiple query calls.

But then again, maybe the solution is to wrap the `loader` function (similar to how its done in `Variables`)

Reply:
Yeah, I wrapped the `loader` function and stuck it on the related hook because I thought those semantics made sense. Feel free to go with what feels most natural for you right now, and we can revisit it later if we think we need a unified approach.